frontend/src/components/shared/FeedbackDialog.tsx

```diff
@@ -0,0 +1,142 @@
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Rating from '@material-ui/lab/Rating';
+import { Locale } from 'scenes/types';
+import { Connection } from 'twilio-client';
+import { PrimaryButton } from './RoundedButton';
+
+interface FeedbackDialogProps extends DialogProps {
+  connection: Connection;
+  locale: Locale;
+}
+
+// Reference for Problem Keys: https://www.twilio.com/docs/voice/sdks/javascript/overview-1x-deprecated/connection
+const EN_STRINGS = {
+  FEEDBACK_DIALOG_TITLE: 'How was your call?',
+  FEEDBACK_SUBMIT_LABEL: 'Submit',
+  FEEDBACK_TITLES: {
+    GOOD: 'GOOD',
+    BAD: 'BAD',
+  },
+  PROBLEM_TITLES: {
+    'Call got disconnected': Connection.FeedbackIssue.DroppedCall,
+    'Lag in audio': Connection.FeedbackIssue.AudioLatency,
+    "Can't hear one another": Connection.FeedbackIssue.OneWayAudio,
+    'Sound breaking up': Connection.FeedbackIssue.ChoppyAudio,
+    'Poor audio quality': Connection.FeedbackIssue.NoisyCall,
+    'Audible echos during call': Connection.FeedbackIssue.Echo,
+  },
+};
+
+const STRINGS = {
+  en: EN_STRINGS,
+  bn: {
+    ...EN_STRINGS,
+  },
+};
+
+// TODO move this to its own module
+const withDialogButtonStyles = withStyles(() => ({
+  root: {
+    padding: '1em 2em',
+    flex: '1 0',
+    margin: '0 0.5rem',
+  },
+}));
+
+const DialogPrimaryButton = withDialogButtonStyles(PrimaryButton);
+
+export default function FeedbackDialog({
+  connection,
+  locale,
+  open,
+  onClose,
+}: FeedbackDialogProps) {
+  const [feedback, setFeedback] = useState(-1);
+  const [hover, setHover] = useState(-1);
+  const [problem, setProblem] = useState<Connection.FeedbackIssue | undefined>(
+    Connection.FeedbackIssue.DroppedCall
+  );
+
+  const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (connection) {
+      // pass problem only when feedback is less than perfect
+      if (feedback === 5) {
+        await connection.postFeedback(feedback);
+      } else {
+        await connection.postFeedback(feedback, problem);
+      }
+    }
+    if (onClose) {
+      onClose({}, 'escapeKeyDown');
+    }
+  };
+
+  return (
+    <Dialog
+      fullWidth
+      disableBackdropClick
+      maxWidth="sm"
+      open={open}
+      onClose={onClose}
+    >
+      <DialogTitle>{STRINGS[locale].FEEDBACK_DIALOG_TITLE}</DialogTitle>
+      <form
+        onSubmit={onSubmit}
+        style={{ display: 'flex', flexDirection: 'column' }}
+      >
+        <div style={{ margin: '1em auto' }}>
+          <Rating
+            size="large"
+            name="hover-feedback"
+            value={feedback}
+            precision={1}
+            onChange={(_event, newValue) =>
+              newValue ? setFeedback(newValue) : setFeedback(-1)
+            }
+            onChangeActive={(_event, newHover) => {
+              setHover(newHover);
+            }}
+          />
+        </div>
+        {feedback < 5 && feedback > 0 ? (
```

Comment:
My preference is not to use magic values like `-1` because it leads to unusual-looking conditions like this. A reader seeing this might wonder 'why this range?' or 'what other values can this value be?'. In fact, I also think that it has also obscured an off-by-one error (did you mean `>= 0` instead of `> 0`?).

This might be made clearer by making feedback a `number | null` and representing a missing value by `null`. This check would then be `feedback !== null & feedback < 5`, which reads in a natural way - if feedback has a specified value and it is less than perfect, we ask for a reason.

Reply:
Ahhh I see! Again I took reference from the material UI documentation haha. Think if we set it to `null` we'll also get the warning of making an uncontrolled variable controlled. Wdyt?

Reply:
If there are no warnings, I think `null` is an easy decision. If there are warnings, the magic value `-1` is ok, but we should document it. To make sure we're on the same page, could you show me the warning?

Reply:
Hmmm, looks like null is fine for the Ratings value 😄